localstack-core/localstack/services/sns/v2/provider.py

```diff
@@ -535,6 +547,121 @@ def list_subscriptions_by_topic(
             response["NextToken"] = next_token
         return response
 
+    #
+    # PlatformApplications
+    #
+    def create_platform_application(
+        self,
+        context: RequestContext,
+        name: String,
+        platform: String,
+        attributes: MapStringToString,
+        **kwargs,
+    ) -> CreatePlatformApplicationResponse:
+        _validate_platform_application_name(name)
+        if platform not in get_valid_platforms():
+            raise InvalidParameterException(
+                f"Invalid parameter: Platform Reason: {platform} is not supported"
+            )
+
+        _validate_platform_application_attributes(attributes)
+
+        # attribute validation specific to create_platform_application
+        if (
+            "PlatformCredential" in attributes.keys()
+            and "PlatformPrincipal" not in attributes.keys()
+        ):
+            raise InvalidParameterException(
+                "Invalid parameter: Attributes Reason: PlatformCredential attribute provided without PlatformPrincipal"
+            )
+
+        elif (
+            "PlatformPrincipal" in attributes.keys()
+            and "PlatformCredential" not in attributes.keys()
+        ):
+            raise InvalidParameterException(
+                "Invalid parameter: Attributes Reason: PlatformPrincipal attribute provided without PlatformCredential"
+            )
+
+        store = self.get_store(context.account_id, context.region)
+        # We are not validating the access data here like AWS does (against ADM and the like)
+        attributes.pop("PlatformPrincipal")
+        attributes.pop("PlatformCredential")
```

Comment:
note for later, just a thoughr: at some point we might want to support sending requests to the 3rd party platforms, but we don't right now anyway 👍 it could be a valuable addition for local development. 

When we will want to implement that we can start storing those credentials somewhere I think